waiter/src/waiter/scaling.clj

```diff
@@ -480,54 +480,51 @@
         apply-scaling-fn (fn apply-scaling-fn [service-id scaling-data]
                            (apply-scaling! executor-multiplexer-chan service-id scaling-data))]
     (async/tap state-mult state-chan)
-    (let [prefix-cid "SCALING"]
-      (async/go
-        (try
-          (loop [{:keys [global-state iter-counter previous-cycle-start-time service-id->router-state
-                         service-id->scale-state timeout-chan] :as current-state}
-                 @state-atom]
-            (reset! state-atom current-state)
-            (let [correlation-id (str prefix-cid "-" iter-counter)
-                  new-state
-                  (timers/start-stop-time!
-                    (metrics/waiter-timer "autoscaler" "iteration")
-                    (let [[args chan] (async/alts! [exit-chan state-chan timeout-chan query-chan] :priority true)]
-                      (condp = chan
-                        exit-chan
-                        (assoc current-state :continue-looping false :timeout-chan nil)
-                        state-chan
-                        (let [{:keys [service-id->healthy-instances service-id->unhealthy-instances service-id->expired-instances]} args
-                              existing-service-ids-set (-> service-id->router-state keys set)
-                              service-ids-set (into (-> service-id->healthy-instances keys set)
-                                                    (-> service-id->unhealthy-instances keys))
-                              deleted-service-ids (set/difference existing-service-ids-set service-ids-set)
-                              new-service-ids (set/difference service-ids-set existing-service-ids-set)
-                              service-id->router-state' (pc/map-from-keys
-                                                          (fn [service-id]
-                                                            {:healthy-instances (count (service-id->healthy-instances service-id))
-                                                             :expired-instances (count (service-id->expired-instances service-id))})
-                                                          service-ids-set)]
-                          (cid/with-correlation-id
-                            correlation-id
+    (let [correlation-id "SCALING"]
+      (cid/with-correlation-id
+        correlation-id
```

Comment:
Nitpick: The let block here seems superfluous.

Reply:
Done.